### PR TITLE
updated to KNP nsp

### PR DESCRIPTION
**Pull request by wadeking98**

### CODE REVIEW

File: openshift/templates/email-verification-agent/email-verification-agent-deploy.yaml [hunk 1]
```diff
@@ -4,48 +4,104 @@ apiVersion: v1
...
-        - - role=service
-          - app=${APP_NAME}${SUFFIX}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+      description: Allow the service to access the agent.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```
Comment (esune) [inline openshift/templates/email-verification-agent/email-verification-agent-deploy.yaml:89]: The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

File: openshift/templates/email-verification-service/email-verification-service-deploy.yaml [hunk 2]
```diff
@@ -4,50 +4,104 @@ apiVersion: v1
...
-      - - role=agent
-        - app=${APP_NAME}${SUFFIX}
-        - env=${TAG_NAME}
-        - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+    description: Allow the agent to access the service.
+    ingress:
+      - from:
+        - podSelector:
+            matchLabels:
+              role: agent
```
Comment (esune) [inline openshift/templates/email-verification-service/email-verification-service-deploy.yaml:90]: The role of the agent can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

File: openshift/templates/postgresql/postgresql-deploy.yaml [hunk 3]
```diff
@@ -6,6 +6,36 @@ metadata:
...
+        app: ${APP_NAME}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the api(s) to access the db.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```
Comment (esune) [inline openshift/templates/postgresql/postgresql-deploy.yaml:24]: The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

**Review by esune — CHANGES_REQUESTED** [review]
Please also include the global `build` NSP template.

File: openshift/templates/email-verification-agent/email-verification-agent-deploy.yaml (hunk 1 above)
Comment (wadeking98) [reply openshift/templates/email-verification-agent/email-verification-agent-deploy.yaml]: change completed

File: openshift/templates/nsp/build.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        name: pods-to-external-network
+        env: ${ENV_NAME}
+    spec:
+      description: |
+        Allow the builds to access the internet.
+        This only needs to be specified once per environment.
+      source:
+        - - $namespace=${NAMESPACE_NAME}-${ENV_NAME}
+      destination:
+        - - ext:name=internet-build-resources
```
Comment (esune) [inline openshift/templates/nsp/build.yaml:54]: These are outdated, we need to use the new KSP-based policies.

**Review by esune — CHANGES_REQUESTED** [review]

File: openshift/templates/nsp/build.yaml (hunk 4 above)
Comment (wadeking98) [reply openshift/templates/nsp/build.yaml]: updated nsp

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Please see comments.  Also I recommend renaming the `postgresql` template folders and files to `db`, to complete the migration from implementation specific naming conventions.

File: openshift/templates/postgresql/postgresql-deploy.yaml [hunk 5]
```diff
@@ -184,6 +222,11 @@ parameters:
     description: Used to group components together.
     required: true
     value: email-verification-service
+  - name: NAMESPACE_NAME
+    displayName: NameSpace name
+    description: name of the project namespace
+    required: true
+    value: myproject
```
[the remark this answers is not in the document]
Comment (wadeking98) [reply openshift/templates/postgresql/postgresql-deploy.yaml]: change completed

File: openshift/templates/postgresql/postgresql-deploy.yaml [hunk 6]
```diff
@@ -44,7 +82,7 @@ objects:
           labels:
             app: ${APP_NAME}${SUFFIX}
             name: ${NAME}${SUFFIX}
-            app-group: ${APP_GROUP}
+            app-group: ${APP_GROUP}
```
[the remark this answers is not in the document]
Comment (wadeking98) [reply openshift/templates/postgresql/postgresql-deploy.yaml]: change completed

File: openshift/TestConnections.txt [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Uses the 'testConnection' script from https://github.com/BCDevOps/openshift-developer-tools
 # Example:
-# echo -e "email-verification-agent\n email-verification-service\n email-verification-demo\n"  | xargs -I {} testConnection -n a99fd4-dev -f TestConnections.txt -p {}
+# echo -e "email-verification-agent\n email-verification-service\n email-verification-demo\n"  | xargs -I {} testConnection -n a99fd4-test -f TestConnections.txt -p {}
```
[the remark this answers is not in the document]
Comment (wadeking98) [reply openshift/TestConnections.txt]: change reverted

**Review by WadeBarnes — APPROVED** [review]

**Review by esune — APPROVED** [review]

**Pull request merged**